README.adoc

```diff
@@ -15,18 +22,543 @@ image:https://circleci.com/gh/edn-query-language/eql/tree/master.svg?style=svg["
+                            {:type         :prop
+                             :dispatch-key :customer/email
+                             :key          :customer/email}]}]}
+----
+
+=== Parameters
+
+EQL properties, joins, and idents have support for parametrization. This allows the
+query to provide an extra dimension of information about the requested data. A parameter
+is expressed by wrapping the thing with an EDN list, like so:
+
+[source,clojure]
+----
+; without params
+[:foo]
+
+; with params
+[(:foo {:with "params"})]
+
+; ast
+
+{:type     :root
+ :children [{:type         :prop
+             :dispatch-key :foo
+             :key          :foo
+             :params       {:with "params"}
+             :meta         {:line 1, :column 15}}]}
+----
+
+Note on the AST side it gets a new `:params` key. Params *must* always be maps, the
+map values can be anything. Here are more examples of parameterizing queries:
+
+[source,clojure]
+----
+; ident with params
+
+[([:ident "value"] {:with "param"})]
+
+{:type     :root
+ :children [{:type         :prop
+             :dispatch-key :ident
+             :key          [:ident "value"]
+             :params       {:with "param"}
+             :meta         {:line 1, :column 15}}]}
+
+; join with params wrap the key with the list
+
+[{(:join-key {:with "params"})
+  [:sub-query]}]
+
+{:type     :root
+ :children [{:type         :join
+             :dispatch-key :join-key
+             :key          :join-key
+             :params       {:with "params"}
+             :meta         {:line 1, :column 16}
+             :query        [:sub-query]
+             :children     [{:type         :prop
+                             :dispatch-key :sub-query
+                             :key          :sub-query}]}]}
+
+; ident join with params
+
+[{([:ident "value"] {:with "params"})
+  [:sub-query]}]
+
+{:type     :root
+ :children [{:type         :join
+             :dispatch-key :ident
+             :key          [:ident "value"]
+             :params       {:with "params"}
+             :meta         {:line 1 :column 16}
+             :query        [:sub-query]
+             :children     [{:type         :prop
+                             :dispatch-key :sub-query
+                             :key          :sub-query}]}]}
+
+; alternate syntax to add params on joins (wrap the entire map, AST result is the same)
+
+[({:join-key
+   [:sub-query]}
+  {:with "params"})]
+
+{:type     :root
+ :children [{:type         :join
+             :dispatch-key :join-key
+             :key          :join-key
+             :params       {:with "params"}
+             :meta         {:line 1, :column 16}
+             :query        [:sub-query]
+             :children     [{:type         :prop
+                             :dispatch-key :sub-query
+                             :key          :sub-query}]}]}
+----
+
+WARNING: remember that unless you are writing on some EDN file, you have to quote the
```

Comment:
NOTE: You'll need to use quote and unquote in CLJ files for calls, otherwise the lists will be evaluated as Clojure calls.  Quote is not necessary in EDN files.